ci-tools/test-matrix/src/html.rs

```diff
@@ -0,0 +1,95 @@
+// Licensed under the Apache-2.0 license
+
+use std::fmt::Write;
+
+use octocrab::models::workflows::Run;
+use serde::Serialize;
+use std::fmt::Debug;
+use tinytemplate::{format_unescaped, TinyTemplate};
+
+use crate::{RunInfo, TestMatrix};
+
+#[derive(Serialize)]
+struct TemplateContext<'a> {
+    pub matrix: &'a TestMatrix,
+    pub javascript: &'static str,
+    pub style: &'static str,
+    pub run: &'a Run,
+    pub run_infos: Vec<MaybeSelected<&'a RunInfo>>,
+}
+
+#[derive(Serialize)]
+pub struct MaybeSelected<T: Serialize> {
```

Comment:
nit: Can you add a comment explaining what it means or an item to be selected? Is this a filtering mechanism?

Reply:
I see this is for the modal with the test output. Still would be good to have a comment explaining this.

Reply:
Done